RELEASE_PROCESS_OPENSEARCH.md

```diff
@@ -0,0 +1,325 @@
+Note: The release managers should be maintainers of a repo under OpenSearch GitHub organization.
+
+#### Release Issue Update
+
+The release issue is created by an [automation workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/releases.yml). Once the release manager is finalized, the release manager should be updating the created release issue. Sample [Release Issue 2.8.0](https://github.com/opensearch-project/opensearch-build/issues/3434)
+
+### Version Increment
+
+Versions are incremented as soon as development starts on a given version to avoid confusion. Following is the example that depicts the version increment.
+
+* OpenSearch: `main` = 2.0.0, `1.x` = 1.1.0, and `1.0` = 1.0.0
+* common-utils: `main` = 1.1.0.0, `1.0` = 1.0.0.0
+
+#### Version Increment Core
+
+To ensure the version incrementation process is handled correctly, it is important to follow up with the core team. Currently, the version incrementation is being done manually by an individual from the core repositories. Sample OpenSearch Version Increment [PR](https://github.com/opensearch-project/OpenSearch/pull/7864/).
+
+#### Release Branch
+
+Not applicable for patch release. This step requires that every team participating in a release has their release branch created for the corresponding release by the date listed on this step. The distribution build job will also start using the release branch to create release candidate instead of `.x` branches.
+
+#### Plugins Version Increment
+
+The next step is to identify the pending plugin version increment pull requests. The creation of plugin version increment pull requests is automated with workflows running in the build repo [OpenSearch Version Increment Workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-increment-plugin-versions.yml),
+[OpenSearch Dashboards Version Increment Workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/osd-increment-plugin-versions.yml). Sample plugin version increment [PR](https://github.com/opensearch-project/job-scheduler/pull/398).
+
+The goal is to get these pull requests merged so that all the plugins are in sync with the core version. It's not always just raising a pull request and getting it merged. There are CI checks for every repo that get triggered when a pull request is raised. These CI checks test against the dependencies. Therefore, before the version increment pull requests are merged, the release manager needs to ensure that the version increment is done for dependencies, built, and pushed to Maven and S3. This is a key step for the release manager to coordinate with the plugin teams, address the dependencies, and get the version increment pull requests merged.
+
+### Distribution Build
```

Comment:
We need to elaborate the steps to create the distribution build. What command to execute, what workflows to invoke, how to include and exclude certain plugins from manifest etc..

Reply:
Let me add some more details to this section.

Reply:
Added some details with a screenshot in my latest commit, please check.